stdpopsim/genomes.py

```diff
@@ -309,89 +316,124 @@ def species_contig(
     def length(self):
         return self.recombination_map.sequence_length
 
-    @property
-    def all_intervals_array(self):
+    def dfe_breakpoints(self):
         """
-        Returns an (n, 3) numpy array with the intervals across all genomic
-        element types in the form (left, right, type).
+        Returns two things: the sorted vector of endpoints of all intervals across
+        all DFEs in the contig, and a vector of integer labels for these intervals,
+        saying which DFE goes with which interval.
+        This provides a complementary method to tell which bit of the contig
+        has which DFE attached, which may be more convenient than the list of two-column
+        arrays provided by interval_list.
+
+        Suppose there are n+1 unique interval endpoints across all the DFE intervals
+        in :property:`.interval_list`.  (If the intervals in that list
+        cover the whole genome, the number of intervals is n.)
+        This method returns a tuple of two things: ``breaks, dfe_labels``.
+        "breaks" is the array containing those n+1 unique endpoints, in increasing order,
+        and "dfe" is the array of length n containing the index of the DFE
+        the applies to that interval. So, ``breaks[0]`` is always 0, and
+        ``breaks[n+1]`` is always the length of the contig, and
+        ``dfe_labels[k] = j`` if
+        ``[breaks[k], breaks[k+1]]`` is an interval in ``contig.interval_list[j]``,
+        i.e., if ``contig.dfe_list[j]`` applies to the interval starting at
+        ``breaks[k]``.  Some intervals may not be covered by a DFE, in which
+        case they will have the label ``-1`` (beware of python indexing!).
```

Comment:
Gah, I wish numpy had an NA type. Any better suggestions than `-1` for this?

Reply:
You could use None, ie. initially create dfe_labels as `np.full(..., None, dtype=object)`. But I'm not sure that's better.

Reply:
Hm - I'd like to have an integer array that we can use for indexing returned, though.

Reply:
I thought that now that a contig starts with a neutral DFE and we can only add new DFEs on top of that all intervals would be covered by a DFE.

Reply:
True, unless you `.clear_dfes()`. (I probably should have just disallowed this... ?)